### ISSUE TITLE
bug: The isDataEqual option has been deprecated and will be removed in the next major version. You can achieve the same functionality by passing a function as the structuralSharing option

### ISSUE THREAD

**Issue by junhuang-ho** [wagmi-dev/wagmi#1221]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.7.9

### Current Behavior

> The isDataEqual option has been deprecated and will be removed in the next major version. You can achieve the same functionality by passing a function as the structuralSharing option

This error/warning keeps spamming the console when I use `useContractRead` with `watch` enabled.

### Expected Behavior

surpress/remove/solve this warning/error.

**Comment by sb777** [comment]
I'm getting this all of a sudden too.

**Comment by digitalnomad91** [comment]
Same here.

**Comment by tmm** [comment]
Thanks for reporting. Happy to pick up this ASAP, but haven't been able to reproduce. Please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) and I'll reopen this issue.

**Comment by sb777** [comment]
Thanks @tmm. It's only happening on my local instance, disappears when I build and deploy. Using NextJS 12 - will try to make a minimal repro later today if I get a chance.

**Comment by sb777** [comment]
Hi @tmm, here are steps to reproduce, let me know if I can add any more information. Thanks for looking at this. I'm having a blast building with wagmi :)

`npm init @rainbow-me/rainbowkit@latest`

in pages/_app.tsx:

comment out alchemyProvider so it doesn't trigger no network error

in pages/index.tsx:

add import
```  
import ERC721 from '<PATH TO OPEN ZEPPELIN>/@openzeppelin/contracts/build/contracts/ERC721.json'
```

under const Home: NextPage = () => {
add
```
const { data: somedata } = useContractRead({
    addressOrName: '0x5d2BF3b4264EFADE95fC89348f1367fCa0552861',
    contractInterface: ERC721.abi,
    functionName: 'ownerOf',
    args: [111],
    watch: true,
  })
```
^ these are arbitrary NFT contract and ID, in this case Starkade


`npm run dev`

and navigate to app, it should trigger a flood of:

<img width="429" alt="Screen Shot 2022-11-03 at 6 07 06 PM" src="https://user-images.githubusercontent.com/10205537/199834350-4f666b8a-fb0d-4c4a-a707-f710090285c2.png">

**Comment by KiSmith777** [comment]
I'm in the same boat. After trying to update to make use of the new scopeKey functionality, which is great btw bc I've been currently running into the same issues that scopeKey addresses, I now have the same error. Did some digging. Looks like the problem lies with the peer deps for @tanstack. 

In the lock file for the latest wagmi release, it shows @tanstack versions 4.10.*, but when installed for end users, it installs versions 4.14.*

**Comment by bigint** [comment]
Happening for me as well with next 13

**Comment by KiSmith777** [comment]
> Happening for me as well with next 13

Its a temp fix but in the meantime, manually installing those dependencies at version 4.10.1 will get rid of the error.

**Comment by ilmpc** [comment]
If you're using pnpm, you can add this to `package.json`
```json
"pnpm": {
    "overrides": {
      "@tanstack/query-sync-storage-persister": "4.13.4",
      "@tanstack/react-query": "4.13.4",
      "@tanstack/react-query-persist-client": "4.13.4"
    }
  }
```

**Comment by farreldarian** [comment]
No issue on `0.7.9` for me, haven't checked what's the last safe version.

**Comment by soccerarenadev** [comment]
Having the same error.

**Comment by linchengzzz** [comment]
have same issue, wagmi current version `0.7.9`

**Comment by danilo-timacum** [comment]
same issue, wagmi 0.7.13

**Comment by ethandaniel** [comment]
I'm having the same issue for wagmmi 0.7.13 using react-typescript with vite. It happens when calling the useContractReads function

**Comment by mtajammulzia** [comment]
I started getting this error today [02 Feb 2023] Wagmi version is ^0.6.0.

**Comment by tmm** [comment]
@mtajammulzia this was fixed in `wagmi@0.7.14`. It's just a warning message so you can upgrade to that version or newer and it should disappear.

**Comment by mtajammulzia** [comment]
Thanks @tmm  that did it. I am also getting another DEPRECTED warning from Zustand package that was installed with Wagmi.
